test/box-luatest/gh_6857_tuple_ext_validation_test.lua

```diff
@@ -0,0 +1,60 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local server = require('test.luatest_helpers.server')
+
+local g = t.group('gh-6857-tuple-ext-validation')
+
+local socket = require('socket')
+local uri = require('uri')
+
+g.before_all = function()
+    g.cluster = cluster:new({})
+    local box_cfg = {
+        log_level = 6,
+    }
+    g.server = g.cluster:build_and_add_server({
+        alias='default',
+        box_cfg = box_cfg,
+    })
+    g.cluster:start()
+    g.listen = uri.parse(server.build_instance_uri('default'))
+    g.server:exec(function()
+        box.schema.space.create('test')
+        box.space.test:create_index('pk',{parts={{1, 'decimal'}}})
+        box.space.test:insert{require('decimal').new(1)}
+    end)
+end
+
+g.after_all = function()
+    g.cluster:drop()
+end
+
+-- Check extension types in tuple handling: previously mp_check() simply
+-- iterated over tuple field headers without looking into the field contents.
+-- This was ok for plain types, like MP_STR, MP_INT and others, because any
+-- msgpack encoded string of any of these types represents a valid value.
+-- This isn't true for extension types, however. Types like decimal, uuid and
+-- others have complex internal structure, which requires additional validation.
+g.test_extension_tuple_validation = function(g)
+    local sock = socket.tcp_connect(g.listen.host, g.listen.service)
+    t.assert_equals(sock:read(9), 'Tarantool', 'Connection established')
+    local data_hex = '19' -- Packet length (25 bytes).
+    data_hex = data_hex .. '8200010100' -- Xrow header. Type = select (0x01)
+    -- Xrow body. Space = 512 (0xcd0200), Index = 0, Limit = 1.
+    data_hex = data_hex .. '8410cd020011001201'
+    -- Key: decimal (malformed)
```

Comment:
Please add a test case for every extension, not just decimal to cover all the code you added.

Reply:
Done